src/RadioButton.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import { color } from 'styled-system'
+import Icon from './Icon'
+import Text from './Text'
+
+const RadioButton = ({
+  className,
+  theme,
+  onClick,
+  isSelected,
+  disabled,
+  radioButtonText
+}) => {
+  const radioIconName = isSelected ? 'radioFilled' : 'radio'
+  const radioIconColor = isSelected ? 'blue' : 'borderGray'
+
+  function handleClick() {
+    if (disabled || isSelected) return
+    onClick()
+  }
+
+  return (
+    <RadioButtonWrap
+      className={className}
+      onClick={handleClick}
+      isSelected={isSelected}
+      disabled={disabled}
+      theme={theme}
+    >
+      <RadioIconWrap>
+        <Icon name={radioIconName} size={24} color={radioIconColor} />
+      </RadioIconWrap>
+      <Text.span pl={4}>{radioButtonText}</Text.span>
+    </RadioButtonWrap>
+  )
+}
+
+const RadioButtonWrap = styled.label`
+  position: relative;
```

Comment:
We should generally avoid setting positioning on the visible elements in design-system components since it would create a new positioning context that would interfere with the consuming app's layout. Instead, it's common to accessibly hide the `<input>` element with something like:

```css
position: absolute;
opacity: 0;
z-index: -1;
```